grype/db/v6/distribution/latest.go

```diff
@@ -67,6 +73,43 @@ func NewLatestFromReader(reader io.Reader) (*LatestDocument, error) {
 	return &l, nil
 }
 
+func NewArchive(path string) (*Archive, error) {
+	tmpDir, err := os.MkdirTemp("", "grype-db-archive")
+	if err != nil {
+		return nil, fmt.Errorf("unable to create temp dir for grype-db archive: %w", err)
+	}
+
+	if err = archiver.Unarchive(path, tmpDir); err != nil {
+		return nil, fmt.Errorf("unable to extract archive: %w", err)
+	}
+
+	cfg := db.Config{
+		DBDirPath: tmpDir,
+	}
+
+	desc, err := db.ReadDescription(cfg.DBFilePath())
+	if err != nil {
+		return nil, fmt.Errorf("failed to calculate description: %w", err)
+	}
+
+	if desc == nil {
+		return nil, fmt.Errorf("unable to describe the database")
+	}
+
+	// calculate the sh256sum of the archive
+	checksum, err := db.CalculateArchiveDigest(cfg.DBFilePath())
```

Comment:
Should the parameter here be `path` not `cfg.DBFilePath()`? It seems weird t calculate the archive digest by doing anything besides hashing the tar.